addons/mail/static/src/components/discuss/discuss_tests.js

```diff
@@ -421,6 +421,175 @@ QUnit.test('sidebar: add channel', async function (assert) {
+                    {id: 20, name: "General", state: 'open', channel_type: 'channel', public: 'public', info: 'unsubscribe'}
+                ];
+                env.services.bus_service.trigger('notification', [notifConfirmUnpin]);
+                return;
+            }
+            if (args.method === 'channel_fold') {
+                assert.step('channel_fold');
+                const notifConfirmFold = [
+                    ["dbName", 'res.partner', 1],
+                    {id: 20, name: "General", state: 'closed', is_minimized: false, channel_type: 'channel', public: 'public'}
+                ];
+                env.services.bus_service.trigger('notification', [notifConfirmFold]);
+            }
+            return this._super(...arguments);
+        },
+    });
+    env = this.env;
+    await afterNextRender(() =>
+        document.querySelector('.o_DiscussSidebar_list .o_DiscussSidebarItem').click()
+    );
+    assert.verifySteps([]);
+    await afterNextRender(() =>
+        document.querySelector('.o_DiscussSidebarItem_commandLeave').click()
+    );
+    assert.verifySteps([
+        'channel_fold',
+        'execute_command'
+    ]);
+    assert.containsNone(
+        document.body,
+        '.o_DiscussSidebar_list .o_DiscussSidebarItem',
+        "The channel must have been removed"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Discuss_thread[data-thread-model="mail.box"][data-thread-id="inbox"]',
+        "The inbox is displayed"
+    );
+});
+
+QUnit.test('sidebar: unpin channel from bus', async function (assert) {
+    assert.expect(6);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: "channel",
+                id: 20,
+                name: "General",
+            }],
+        },
+    });
+
+    await this.start();
+    assert.strictEqual(
+        document.body.querySelector('.breadcrumb-item.active').textContent,
+        'Inbox',
+        "The inbox is displayed"
+    );
+    assert.containsOnce(document.body, '.o_DiscussSidebar_list .o_DiscussSidebarItem', "1 channel is present");
+    assert.strictEqual(
+        document.querySelector(`.o_DiscussSidebar_list .o_DiscussSidebarItem .o_DiscussSidebarItem_name`).textContent,
+        'General',
+        "The channel name is 'General'"
+    );
+
+    await afterNextRender(() =>
+        document.querySelector('.o_DiscussSidebar_list .o_DiscussSidebarItem').click()
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Discuss_thread[data-thread-model="mail.channel"][data-thread-id="20"]',
+        "The channel #General is displayed"
+    );
+
+    const notif = [
+        ["dbName", 'res.partner', 1],
+        {id: 20, name: "General", state: 'open', channel_type: 'channel', public: 'public', info: 'unsubscribe'}
+    ];
+    await afterNextRender(() =>
+        this.env.services.bus_service.trigger('notification', [notif])
+    );
```

Comment:
- move `notif` in afterNextRender inner function scope
- multi-line for objects with several keys (and it doesn't use ES6 short syntax)
- put a comment for what this simulates
- `1` is a magic number that hardly means something here. It should use `this.env.messaging.currentPartner.id` (because we could assume messaging notification handler could ignore notifications addressed to someone else).